{
  "comments": [
    {
      "key": {
        "uuid": "68662314_d8f0ed0b",
        "filename": "subcmds/upload.py",
        "patchSetId": 3
      },
      "lineNbr": 429,
      "author": {
        "id": 1006257
      },
      "writtenOn": "2013-12-12T00:23:56Z",
      "side": 1,
      "message": "This should be moved down after this if/else because you should return after printing the error.",
      "revId": "b69cc3a8fda3535a4cae2f68f07bdfb33e6ddc89",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "88343708_61ecb63a",
        "filename": "subcmds/upload.py",
        "patchSetId": 3
      },
      "lineNbr": 429,
      "author": {
        "id": 1014507
      },
      "writtenOn": "2013-12-12T06:47:50Z",
      "side": 1,
      "message": "I don\u0027t know if I understand what you want me to do here.\n\nI was trying to keep the same logic as before, which is that \"no branches ready for upload\" would be printed (in addition to my new error message) if avail is None. If I return after the error message, that wouldn\u0027t happen.",
      "parentUuid": "68662314_d8f0ed0b",
      "revId": "b69cc3a8fda3535a4cae2f68f07bdfb33e6ddc89",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "88343708_8139faf2",
        "filename": "subcmds/upload.py",
        "patchSetId": 3
      },
      "lineNbr": 429,
      "author": {
        "id": 1006257
      },
      "writtenOn": "2013-12-12T21:08:27Z",
      "side": 1,
      "message": "But the current logic is wrong, isn\u0027t it?  The issue you filed said you got a traceback:\nhttps://code.google.com/p/git-repo/issues/detail?id\u003d156\n\nWe don\u0027t want a traceback, just a clean error message.",
      "parentUuid": "88343708_61ecb63a",
      "revId": "b69cc3a8fda3535a4cae2f68f07bdfb33e6ddc89",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "48579f41_17e25447",
        "filename": "subcmds/upload.py",
        "patchSetId": 3
      },
      "lineNbr": 429,
      "author": {
        "id": 1014507
      },
      "writtenOn": "2013-12-12T23:44:48Z",
      "side": 1,
      "message": "I think the logic is correct.\n\nThe backtrace happened below on line 458. The original code had avail \u003d\u003d [None] if cbr was set but not uploadable. The \"if avail:\" on line 437 below would then be true and [None] would be appended to pending, making [[None]] in the case of pushing the current branch. As a result of that, the \"if pending\" check on 440 would be successful and run the repo hooks regardless of whether the current branch was pushable or not. Then, on line 456, \"if not pending\" would be false, falling to the elif on line 458. The len(pending) \u003d\u003d 1 would be true for [[None}]. Then pending[0][1][0] would be expanded. However, pending[0] is None, not an array, so the second index lookup (the \"[1]\") would fail.\n\nIn the code in the lastest patchset, the avail is properly set to None (instead of [None]), so the test \"if avail:\" on line 437 fails, thus not appending to the pending array. The hooks will not be run, and line 456\u0027s \"if not pending\" will be true and thus run the print on line 457 (\"no branches ready for upload\").",
      "parentUuid": "88343708_8139faf2",
      "revId": "b69cc3a8fda3535a4cae2f68f07bdfb33e6ddc89",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "08c2071e_0de71158",
        "filename": "subcmds/upload.py",
        "patchSetId": 3
      },
      "lineNbr": 430,
      "author": {
        "id": 1006257
      },
      "writtenOn": "2013-12-12T00:23:56Z",
      "side": 1,
      "message": "stick with single quotes to match the surrounding context.",
      "revId": "b69cc3a8fda3535a4cae2f68f07bdfb33e6ddc89",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "88343708_81e73a56",
        "filename": "subcmds/upload.py",
        "patchSetId": 3
      },
      "lineNbr": 430,
      "author": {
        "id": 1014507
      },
      "writtenOn": "2013-12-12T06:47:50Z",
      "side": 1,
      "message": "Done",
      "parentUuid": "08c2071e_0de71158",
      "revId": "b69cc3a8fda3535a4cae2f68f07bdfb33e6ddc89",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153",
      "unresolved": false
    }
  ]
}